### PR TITLE
spec/support/server.rb Refactor plus raise condition fix while running test in a logstash version.

### PR DESCRIPTION
**Pull request by purbon**
Refactored the server support files into his own more specific location, also fixed a raise condition that happened while running the test within a logstash clone.

Fix a minor spec style.

### CODE REVIEW

File: spec/support/server.rb [hunk 1]
```diff
@@ -12,8 +12,10 @@ def start
       Thread.new do
         client = @server.accept
         while true
-          line = client.readline
-          @queue << line
+          if !client.eof?
+            line = client.readline
+            @queue << line
+          end
         end
```
Comment (colinsurprenant) [inline spec/support/server.rb:19]:
``` ruby
Thread.new do
  client = @server.accept
  @queue << client.readline while !client.eof?
end
```

?

Comment (purbon) [inline spec/support/server.rb:19]: better :-)

**Comment by ph** [comment]
LGTM

File: spec/support/server.rb (hunk 1 above)
Comment (ph) [inline spec/support/server.rb:19]:
``` ruby
@queue << client.readline until client.eof?
```

Comment (purbon) [inline spec/support/server.rb:19]: not a big fun of until :-)

**Pull request closed without merging**